tests/new_core/processors/test_metric_calc.py

```diff
@@ -0,0 +1,364 @@
+"""
+Units tests for the `metric_calc` processor.
+
+This processor calculates various statistical metrics (e.g., mean, median, percentiles)
+over specified dimensions of the input data.
+
+The tests validate the correct functionality of the processor, including handling of
+different metrics, percentiles, dimensions, and edge cases.
+"""
+
+from typing import Any, Dict
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+import xarray as xr
+
+from climakitae.core.constants import _NEW_ATTRS_KEY, UNSET
+from climakitae.new_core.processors.metric_calc import MetricCalc
+
+
+@pytest.fixture
+def test_da():
+    """Set up a sample xarray DataArray for testing."""
+    data = xr.DataArray(
+        np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+        dims=["time", "lat"],
+        coords={"time": [0, 1, 2], "lat": [10, 20, 30]},
+        name="t2max",
+    )
+    yield data
```

Comment:
For future reference you dont _need_ to use `yield` unless you have teardown to do, but it's good to be consistent and we use it elsewhere, so this is fine.